lms/djangoapps/django_comment_client/utils.py

```diff
@@ -819,14 +818,23 @@ def get_group_id_for_comments_service(request, course_key, commentable_id=None):
             _verify_group_exists(group_id, course_discussion_settings)
         else:
             # regular users always query with their own id.
-            group_id = get_group_id_for_user(request.user, course_discussion_settings)
+            group_id = get_group_id_for_user_from_cache(request.user, course_key)
         return group_id
     else:
         # Never pass a group_id to the comments service for a non-divided
         # commentable
         return None
 
 
+@request_cached
+def get_group_id_for_user_from_cache(user, course_id):
+    """
+    Caches the results of get_group_id_for_user, but serializes the course_id
+    instead of the course_discussions_settings object as cache keys.
+    """
+    return get_group_id_for_user(get_course_discussion_settings(course_id))
```

Comment:
are we using the `user` arg in the body? should we pass it in `get_group_id_for_user`?

Reply:
Whoops, went too fast here, good catch!